src/main/java/org/isf/accounting/gui/PatientBillEdit.java

```diff
@@ -471,50 +544,64 @@ private void checkBill() {
 			if (patient != null) {
 				setPatientSelected(patient);
 				Admission currentAdmission = admissionBrowserManager.getCurrentAdmission(patient);
-				
+
 				Icon icon = UIManager.getIcon("OptionPane.warningIcon");
 				if (thisBill.getAdmission() == null && currentAdmission != null) {
-					int ok = MessageDialog.yesNo(PatientBillEdit.this, icon, "angal.newbill.thispatientisadmittednowdoyouwanttolinkthisbilltothecurrentadmission.msg");
+					int ok = MessageDialog.yesNo(PatientBillEdit.this, icon,
+									"angal.newbill.thispatientisadmittednowdoyouwanttolinkthisbilltothecurrentadmission.msg");
 					if (ok == JOptionPane.OK_OPTION) {
 						thisBill.setAdmission(currentAdmission);
 						modified = true;
 					}
 				}
 				if (thisBill.getAdmission() != null && currentAdmission == null) {
-					int ok = MessageDialog.yesNo(PatientBillEdit.this, icon, "angal.newbill.thispatientisnolongeradmitteddoyouwanttounlinkthisbillfromthepreviousadmission.msg");
+					int ok = MessageDialog.yesNo(PatientBillEdit.this, icon,
+									"angal.newbill.thispatientisnolongeradmitteddoyouwanttounlinkthisbillfromthepreviousadmission.msg");
 					if (ok == JOptionPane.OK_OPTION) {
 						thisBill.setAdmission(currentAdmission);
```

Comment:
Is this line right?  `currentAdmission` is `null`; see line 557

Reply:
yes, if the user accepts the admission connected to the bill should be set to currentAdmission=null.
basically line 557 is just to prompt the right question, but maybe the logic can be improved